### PR TITLE
SmartSport guide completed

### PR DESCRIPTION
**Pull request by Atlantido**

### CODE REVIEW

File: docs/sensors/smartspot.md [hunk 1]
```diff
@@ -1,11 +1,199 @@
...
-## Register your SmartSpot in the IoT Agent
+In order to monitorize the air pollution, Smart Spot measures the NO2, CO, SO2 and O3 at specific points in real time. We have a sophisticated high precision lab, with a Mass Flow Controller and a Zero Air Generator, where through Machine Learning algorithms we improve the precision of our sensor measures reducing the effect of cross sensitivity.
 
-## Register your Air Quality Dashboard in the SmartSpot
+For the care and management of data collected by the Smart Spot a maintenance platform is needed, our device can be managed remotely by platforms which use OMA LWM2M as FIWARE. We deploy FIWARE to manage Smart Spot data and also, we integrate our solution in already existing FIWARE ecosystems.
+
+Optionally, this system provides information about crowds in specific areas, detecting people with smartphone WiFi switched-on.
+
+
+## **What do you need?**
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: please use canonical markdown

File: docs/sensors/smartspot.md [hunk 2]
```diff
@@ -1,11 +1,199 @@
-# Creating your own SmartSpot for Air quality
+# User&#39;s guide for Smart Spot Starter Kit
 
-## What you need to order
+## **Introduction**
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: please use canonical markdown

File: docs/sensors/smartspot.md [hunk 3]
```diff
@@ -1,11 +1,199 @@
...
 
-## How to install and configure it
+In order to offer a simple and intuitive way to get started with our Smart Spot we have developed an Open Source solution called Smart Spot Starter Kit, which has an obvious educational purpose.  Just by simply obtaining this kit and follow this guide you will be able to manage the device and to acquire data from its board sensors, apart from the ones you connect, with your smartphone.
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: apply linter, lines should me max 80 chars, see for applied rules:
https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md

and here for the linter detected issues: https://travis-ci.org/smartsdk/guided-tour/jobs/346396767

File: docs/sensors/smartspot.md [hunk 4]
```diff
@@ -1,11 +1,199 @@
...
+In order to monitorize the air pollution, Smart Spot measures the NO2, CO, SO2 and O3 at specific points in real time. We have a sophisticated high precision lab, with a Mass Flow Controller and a Zero Air Generator, where through Machine Learning algorithms we improve the precision of our sensor measures reducing the effect of cross sensitivity.
 
-## Register your Air Quality Dashboard in the SmartSpot
+For the care and management of data collected by the Smart Spot a maintenance platform is needed, our device can be managed remotely by platforms which use OMA LWM2M as FIWARE. We deploy FIWARE to manage Smart Spot data and also, we integrate our solution in already existing FIWARE ecosystems.
+
+Optionally, this system provides information about crowds in specific areas, detecting people with smartphone WiFi switched-on.
+
+
+## **What do you need?**
+
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: where can it be ordered?

imagine people really need to build the system, how that will be possible?

File: docs/sensors/smartspot.md [hunk 5]
```diff
@@ -1,11 +1,199 @@
...
+### What is included with The Smart Spot Starter Kit:
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: this level three is not needed given that there are no other level three in same section

File: docs/sensors/smartspot.md [hunk 6]
```diff
@@ -1,11 +1,199 @@
...
+
+1. ESP32 DevKitC.
+
+![ESP32 DevKitc](./images/ESP32-DevKitC.png)
+
+2. Starter Kit firmware
+
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: use canonical md, start always with 1. and use proper indentation to ensure numbering is accounted correctly

File: docs/sensors/smartspot.md [hunk 7]
```diff
@@ -1,11 +1,199 @@
...
+
+## **What do you need?**
+
+### What is included with The Smart Spot Starter Kit:
+
+1. ESP32 DevKitC.
+
+![ESP32 DevKitc](./images/ESP32-DevKitC.png)
+
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: picture should be intended

File: docs/sensors/smartspot.md [hunk 8]
```diff
@@ -1,11 +1,199 @@
...
+	* Luminosity Sensor
+	* RGB led
+	* GPIO Led
+
+4. Micro USB Cable
+5. Smart Spot Air Quality Expansion Board
+6. I2C Cable
+ 
+![I2C Cable](./images/I2C.png)
+
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: picture should be indented

File: docs/sensors/smartspot.md [hunk 9]
```diff
@@ -1,11 +1,199 @@
...
+7. 4 x Gas Sensors:
+ 	* Sulfur Dioxide (Red).
+ 	* Ozone + Nitrogen Dioxide (Yellow).
+ 	* Carbon Monoxide (Green).
+ 	* Nitrogen Dioxide (Orange).
+
+![Gas Sensor](./images/sensor.png)
+
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: picture should be indented

File: docs/sensors/smartspot.md [hunk 10]
```diff
@@ -1,11 +1,199 @@
...
+You will need a PC loaded with Windows, Linux or Mac OS with internet connection in order to download the Starter Kit firmware and the ESP32 toolchain and ESP-IDF.
+
+## **Getting Started**
+
+This chapter is a guide for Windows users. In case you use another OS, you can find further information in the following link
+
+[http://esp-idf.readthedocs.io/en/latest/get-started/index.html](http://esp-idf.readthedocs.io/en/latest/get-started/index.html)
+
+
+1) First, you need to download the ESP32 toolchain by clicking in this link: [https://dl.espressif.com/dl/esp32\_win32\_msys2\_environment\_and\_toolchain-20180110.zip](https://dl.espressif.com/dl/esp32_win32_msys2_environment_and_toolchain-20180110.zip). Unzip it and place somewhere safe. This guide assumes the environment is placed in _C:\ _ .
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: use canonical md numbered list.

```
1. xxx
```

File: docs/sensors/smartspot.md [hunk 11]
```diff
@@ -1,11 +1,199 @@
...
+
+2) You will also need the ESP32 API. In order to download it, open a terminal, navigate to a directory where you want to place the IDF and use these commands:
+
+~~~
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: use canonical md code and indent it for proper alignment below the list

1. xxxx

    ```bash
    $ git clone --recursive https://github.com/espressif/esp-idf.git
 
    $ git checkout tags/v3.0-rc1
 
    $ git submodule update –init
    ```

File: docs/sensors/smartspot.md [hunk 12]
```diff
@@ -1,11 +1,199 @@
...
+
+$ git clone --recursive https://github.com/espressif/esp-idf.git
+
+$ git checkout tags/v3.0-rc1
+
+$ git submodule update –init
+
+~~~
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: see above

File: docs/sensors/smartspot.md [hunk 13]
```diff
@@ -1,11 +1,199 @@
...
+
+Every time you restart your PC you will need to define the IDF\ _PATH by using this command:
+
+~~~
+
+$ export IDF\_PATH=&quot;C:/msys32/home/user-name/esp/esp-idf&quot;
+
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: see above

File: docs/sensors/smartspot.md [hunk 14]
```diff
@@ -1,11 +1,199 @@
...
+In case you want to set the path permanently, check out this link:
+
+[http://esp-idf.readthedocs.io/en/latest/get-started/add-idf\_path-to-profile.html#add-idf-path-to-profile-windows](http://esp-idf.readthedocs.io/en/latest/get-started/add-idf_path-to-profile.html#add-idf-path-to-profile-windows)
+
+3) Now is time to download the Smart Sport Firmware. You should open a terminal in another directory and clone our firmware by using the following command:
+
+~~~
+
+$ git clone –-https://github.com/HOP-Ubiquitous/SmartSpot\_SmartSDK\_Firmware.git
+
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: see above

File: docs/sensors/smartspot.md [hunk 15]
```diff
@@ -1,11 +1,199 @@
...
+You may have to download its driver in case you are not able to flash. Download it here.
+
+[https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers](https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers)
+
+6) Copy the port and the directory of bootloader.bin, smartspot-esp32.bin, and partitions\_singleapp.bin. Replace it by the bolded words of the following command. Use it to flash the esp32.
+
+~~~
+
+$ python /c/Users/HOPU/GitHub/espidf/components/esptool\_py/esptool/esptool.py --chip esp32 --port **COMX** --baud 115200 --before default\_reset --after hard\_reset write\_flash -z --flash\_mode dio --flash\_freq 40m --flash\_size detect 0x1000 **C:/Users/HOPU/GitHub/SmartSpot\_SmartSDK\_Firmware/bootloader.bin** 0x10000
+
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: see above

File: docs/sensors/smartspot.md [hunk 16]
```diff
@@ -1,11 +1,199 @@
...
+**C:/Users/HOPU/GitHub/SmartSpot\_SmartSDK\_Firmware/smartspot-esp32.bin** 0x8000 **C:/Users/HOPU/GitHub/SmartSpot\_SmartSDK\_Firmware/partitions\_singleapp.bin**
+
+~~~
+
+![Commands](./images/python.png)
+
+A message like the one above should appear if you flashed the firmware successfully.
+
+## **Expansion board integration**
+
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: use canonical md

File: docs/sensors/smartspot.md [hunk 17]
```diff
@@ -1,11 +1,199 @@
...
+SSID: **defaultSSAP**
+
+Password: **defaultSSAP1234**
+
+You can easily share the connection with your iOS or Android.
+
+![Connection](./images/connection.png)
+
+## **Fiware integrations**
+FIWARE ([www.fiware.org](http://www.fiware.org)) is an open platform promoted by the European Commision and maintained by the FIWARE Foundation, where HOP Ubiquitous is Gold Member.
```
Comment (chicco785) [inline docs/sensors/smartspot.md]: this is a tutorial, so this text is not appropriate,
just shortly describe how to connect the sensor to the proper IoT Agent.

you should also describe how to register a url to be use for physical web interaction

**Review by chicco785 — CHANGES_REQUESTED** [review]

File: docs/sensors/smartspot.md [hunk 18]
```diff
@@ -1,11 +1,199 @@
...
+ 	* Sulfur Dioxide (Red).
+ 	* Ozone + Nitrogen Dioxide (Yellow).
+ 	* Carbon Monoxide (Green).
+ 	* Nitrogen Dioxide (Orange).
+
+![Gas Sensor](./images/sensor.png)
+
+You will need a PC loaded with Windows, Linux or Mac OS with internet connection in order to download the Starter Kit firmware and the ESP32 toolchain and ESP-IDF.
+
+## **Getting Started**
```
Comment (taliaga) [inline docs/sensors/smartspot.md]: I'd suggest adding (windows only) here

File: docs/sensors/smartspot.md [hunk 19]
```diff
@@ -1,11 +1,199 @@
...
+* O3: Ozone
+* CO: Carbon monoxide
+
+![Sensors](./images/sensors.png)
+
+In order to carry out the connection between the expansion board and the Smart Spot Air Quality Expansion Board you only have to plug the i2c cable in both sides. You can use any i2c port you want since they all have the same behavior. Finally, plug the gas sensors in the i2c board.
+
+## **Register your Smart Spot in the IoT Agent**
+
+First, you need to create a WiFi AP with your smartphone for the first connection of the SmartSpot. Once you register in the IoT Agent, you can change the WiFi parameters. This are the default SSID and password:
```
Comment (taliaga) [inline docs/sensors/smartspot.md]: There are missing (assumed) steps here, please either add info or links to where that is documented (create ap, register iota, etc)

**Review by taliaga — CHANGES_REQUESTED** [review]

**Comment by dancn** [comment]
Hello,
small suggestion: for the MarkDown compliance checking you can use this one-liner:
https://github.com/smartsdk/guided-tour/pull/4/files#diff-04c6e90faac2675aa89e2176d2eec7d8R29

**Comment by taliaga** [comment]
Hi @Atlantido, content-wise, the changes LGTM.
Last thing I'd suggest is having a look a the CI issues (https://travis-ci.org/smartsdk/guided-tour/jobs/364016350).
Thanks!

**Comment by chicco785** [comment]
@Atlantido please lint the markdown as by CI report https://travis-ci.org/smartsdk/guided-tour/jobs/364016350

thx

**Comment by Atlantido** [comment]
@chicco785 we are adding new content to guide. When we finish adding the new content then we will do pull request

**Comment by chicco785** [comment]
hi, by when? we want this to be online before the summit next week :)

thx

**Comment by Atlantido** [comment]
Tomorrow is a holiday in Spain and this Wednesday in the afternoon or first thing on Monday we will do the pull request

File: docs/sensors/smartspot.md [hunk 20]
```diff
@@ -1,11 +1,982 @@
...
+      available. The specified port in the server section, needs to be the
+      same here.
+
+In the following table you can find a config.js file as example for the FIWARE
+docker infrastructure that can be found in the [HOP Ubiquitous
+gitHub](https://github.com/HOP-Ubiquitous/fiware-docker-infrastructure). In the
+readme file of that repository, a complete guide to deploy the docker
+infrastructure can be found.
+
+~~~
```
Comment (taliaga) [inline docs/sensors/smartspot.md]: As promised, I'm checking the latest changes. Some comments here:
- I'd put this huge config in a file and reference from it. It bloats the document unnecessarily. If the config already lives somewhere else, just reference it.
- Use single backticks to wrap special words such as environment variable names or filenames: `config.js`
- Minor, but I've never seen those ~~~ to indicate code blocks, I knew of ``` . Does that render fine in the generated docs?. (see a pointer in the README at the root of this project for the doc builder tool)

File: docs/sensors/smartspot.md [hunk 21]
```diff
@@ -1,11 +1,982 @@
...
+          The IOT-Agent will set a observer in every active resource after the
+          device connection.
+        * lwm2mResourceMapping: This section is the one that is going to set
+          the needed OMA-LwM2M resources to be read in order to map the lazy
+          and active resources.
+    * providerURL: The URL in which one the LwM2M IOT-Agent API is going to be
+      available. The specified port in the server section, needs to be the
+      same here.
+
+In the following table you can find a config.js file as example for the FIWARE
```
Comment (taliaga) [inline docs/sensors/smartspot.md]: table?

File: docs/sensors/smartspot.md [hunk 22]
```diff
@@ -1,11 +1,982 @@
...
+    providerUrl: 'http://iotagent:4042',
+    deviceRegistrationDuration: 'P1M'
+};
+
+module.exports = config;
+~~~
+
+### Build architecture through docker-compose
+
+Please, take a look to the docker-compose.yml file, in this one you can find the
```
Comment (taliaga) [inline docs/sensors/smartspot.md]: Readers will not know where that file is, will they?

File: docs/sensors/smartspot.md [hunk 23]
```diff
@@ -1,11 +1,982 @@
...
+
+The docker-compose service description do not use version tag in the majority of
+the services. That implies the use of the service last version at the time of
+perform the pull. There are some exceptions like the
+[MongoDB](https://www.mongodb.com) version or the IoT Agent that must be build
+in the host machine.
+
+## To Be Done
+
+Add FIWARE security stack Create different docker-composes adapting the
```
Comment (taliaga) [inline docs/sensors/smartspot.md:576]: may read better as a list (or bracketed list) as below:
- AA
- BB
- [ ] A
- [ ] B

**Review by taliaga — CHANGES_REQUESTED** [review]

**Comment by Atlantido** [comment]
@taliaga In your requested changes you said "may read better as a list (or bracketed list)". I don´t understand what you mean with it can be read better as a list.

**Review by chicco785 — APPROVED** [review]

**Pull request merged**